--- a/src/steamcompmgr.cpp
+++ b/src/steamcompmgr.cpp
@@ -2324,11 +2324,11 @@
 	frameInfo.allowVRR             = false;
 	frameInfo.bFadingOut           = false;
 
-	// Apply screenshot-style color management.
+	// Apply display color management LUTs (includes gamut wideness / night mode from Steam sliders).
 	for ( uint32_t nInputEOTF = 0; nInputEOTF < EOTF_Count; nInputEOTF++ )
 	{
-		frameInfo.lut3D[nInputEOTF]     = g_ScreenshotColorMgmtLuts[nInputEOTF].vk_lut3d;
-		frameInfo.shaperLut[nInputEOTF] = g_ScreenshotColorMgmtLuts[nInputEOTF].vk_lut1d;
+		frameInfo.lut3D[nInputEOTF]     = g_ColorMgmtLuts[nInputEOTF].vk_lut3d;
+		frameInfo.shaperLut[nInputEOTF] = g_ColorMgmtLuts[nInputEOTF].vk_lut1d;
 	}
 
 	const uint64_t ulFocusAppId = s_pPipewireBuffer->gamescope_info.focus_appid;
